src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -283,21 +345,60 @@ private void validatePartitionProperties(@Nullable Schema schema, FailureCollect
                                  "exists without partitioning. Please verify the partitioning configuration.",
                                table.getTableId().getTable()));
       }
-      if (timePartitioning != null && timePartitioning.getField() != null
-        && !timePartitioning.getField().equals(partitionByField)) {
-        collector.addFailure(String.format("Destination table '%s' is partitioned by column '%s'.",
-                                           table.getTableId().getTable(),
-                                           timePartitioning.getField()),
-                             String.format("Set the partition field to '%s'.", timePartitioning.getField()))
-          .withConfigProperty(NAME_PARTITION_BY_FIELD);
+      RangePartitioning rangePartitioning = tableDefinition.getRangePartitioning();
+      if (timePartitioning == null && rangePartitioning == null && shouldCreatePartitionedTable()) {
+        LOG.warn(String.format(
```

Comment:
auto-create properties are a little funny because they only apply once. I don't remember the exact reason we decided this shouldn't be an error, but its good to follow the existing behavior.

In general, we should have failures for anything that will never be correct, like having a negative number when it's not allowed. We're more lax about things that can be fixed, like if a GCS path doesn't exist when we expect it to, as we don't want pipeline deployment to be dependent on having the environment set up. This category of potential problems fits better as a warning instead of a failure, though we don't currently have a way to surface these types of things anywhere except the logs.